packages/solidity-contracts/scripts/hardhat/eventFilter.ts

```diff
@@ -0,0 +1,46 @@
+import { task } from 'hardhat/config';
+import { HardhatRuntimeEnvironment } from 'hardhat/types';
+import { writeFileSync } from 'fs';
+
+task(
+  'grant-role-event-filter',
+  'Filters grant role event for a specific contract to keep track of assigned roles'
+)
+  .addParam('contract', 'address of the contract')
+  .setAction(
+    async (taskArgs: any, hre: HardhatRuntimeEnvironment): Promise<void> => {
+      const provider = new hre.ethers.JsonRpcProvider(process.env.RPC_URL);
+
+      const grantRoleEvenABI = [
+        'event RoleGranted(bytes32 indexed role, address indexed account, address indexed sender)',
+      ];
+
+      const contract = new hre.ethers.Contract(
+        taskArgs.contract,
+        grantRoleEvenABI,
+        provider
+      );
+
+      const eventPayload: any = [];
+
+      try {
+        const events = await contract.queryFilter(
+          contract.filters.RoleGranted()
```

Comment:
What happens when you query too many blocks?

Reply:
currently tested with a public rpc, triggering the ci frequently might result in rate limit issues, as currently we query all blocks.

With a private rpc we should be good, but I was also thinking if we can specify a block range too but for now decided to keep the number of inputs minimal, so didn't add that

Reply:
imo, hardhat tasks are not the ideal way to do this. Also why do we need this to be in the ci?

Reply:
we need to be able to manually trigger it and there is no tool with a ui live currently that we can use for filtering events, there is a rate limit error risk with the script approach but using a pvt rpc reduces that risk to a great extent

Reply:
> imo, hardhat tasks are not the ideal way to do this. Also why do we need this to be in the ci?

Please propose alternatives. I am tempted to use external infra for this task, like envio.

Reply:
I guess we can use the graph protocol or quickly spin up our own indexer using ponder.sh